lib/operations/create_indexes.js

```diff
@@ -2,34 +2,55 @@
 
 const Aspect = require('./operation').Aspect;
 const defineAspects = require('./operation').defineAspects;
-const OperationBase = require('./operation').OperationBase;
-const executeCommand = require('./db_ops').executeCommand;
+const CommandOperationV2 = require('./command_v2');
 const MongoError = require('../core').MongoError;
+const maxWireVersion = require('../core/utils').maxWireVersion;
 const ReadPreference = require('../core').ReadPreference;
 
-class CreateIndexesOperation extends OperationBase {
+/**
+ * Creates one or more indexes on the collection.
+ *
+ * @class
+ */
+class CreateIndexesOperation extends CommandOperationV2 {
+  /**
+   * Construct a CreateIndexes operation.
+   *
+   * @param {Collection} collection parent collection
+   * @param {string|object} indexSpecs Defines the indexes.
+   * @param {object} [options] Optional settings. See Collection.prototype.createIndexes for a list of options.
+   */
   constructor(collection, indexSpecs, options) {
-    super(options);
-
+    super(collection, options);
     this.collection = collection;
     this.indexSpecs = indexSpecs;
   }
 
-  execute(callback) {
-    const coll = this.collection;
+  /**
+   * Execute the operation.
+   *
+   * @param {any} server
+   * @param {Collection~resultCallback} [callback] The command result callback
+   */
+  execute(server, callback) {
+    const options = Object.assign({}, this.options, { readPreference: ReadPreference.PRIMARY });
```

Comment:
We no longer need to explicitly add a primary read preference if one wasn't provided. Since you're using the `CommandOperationV2` a fully resolved `readPreference` is available directly on the operation itself. Additionally, server selection has already occurred by the time this method is called, so there's no need to worry about read preferences at all.